tests/backends/base/test_base.py

```diff
@@ -60,6 +61,28 @@ def test_check_database_version_supported_with_none_as_database_version(self):
         with patch.object(connection.features, "minimum_database_version", None):
             connection.check_database_version_supported()
 
+    def test_release_memory_without_garbage_collection(self):
+        # Clear the garbage list to avoid interfering with the test.
+        gc.collect()
+        # Turn off automatic garbage collection.
+        gc.disable()
+        # Set the garbage collection debugging flag to store all unreachable
+        # objects in `gc.garbage`.
+        gc.set_debug(gc.DEBUG_SAVEALL)
+
+        # Create and close a connection.
+        connection.copy().close()
+
+        # Enforce garbage collection.
```

Comment:
Should we check here that we do have garbage at all? Otherwise the collection won't do much? Also reading the docs at https://docs.python.org/3/library/gc.html#gc.garbage about `gc.garbage` it seems to suggest that it would be empty either way (or would `DEBUG_SAVEALL` change that?).

Reply:
`gc.garbage` seems to be empty before garbage collection as well (which I think is okay if python immediately clears the objects?) -- but that begs the question if we need to enforce garbage collection at all here -- wouldn't it be enough that `gc.garbage` is empty?

Reply:
> `gc.garbage` is only populated if debugging flags are set because doing so costs extra CPU cycles. 

Ah, now it clicked. Thank you for the extended explanation.

> I think it’s worth describing why all the steps are necessary for the test to be meaningful. Otherwise tomorrow someone could remove one of the steps and the test would still seem to be passing (without actually testing anything).

Absolutely, this is exactly why am I asking my questions to ensure that the test actually tests what it is supposed to :D